tests/tests_unit/test_data_classes/test_capabilities.py

```diff
@@ -215,122 +215,77 @@ def test_load__action_does_not_exist(self, acl_cls_name: str, bad_action: str, d
-                                scope=RawAcl.Scope.Table(
-                                    {"my_db": DatabaseTableScope("my_db", ["my_table", "my_other_table"])}
-                                ),
-                            ),
-                            project_scope=ProjectCapability.Scope.All(),
-                        )
-                    ]
-                ),
-                RawAcl(
-                    [RawAcl.Action.Read],
-                    scope=RawAcl.Scope.Table({"unknown_db": DatabaseTableScope("unknown_db", ["my_table"])}),
+            ProjectCapability(
+                capability=RawAcl(
+                    [RawAcl.Action.Read], scope=RawAcl.Scope.Table({"my_db": ["my_table", "my_other_table"]})
                 ),
-                False,
+                project_scope=ProjectCapability.Scope.Projects([project_name]),
             ),
+        ]
+    )
+
+
+class TestProjectCapabilitiesList:
+    def test_project_scope_is_all_projects(self, proj_cap_allprojects_dct):
+        loaded = ProjectCapability.load(proj_cap_allprojects_dct.copy())
+        assert type(loaded.project_scope) is AllProjectsScope
+
+        loaded = ProjectCapabilitiesList.load([proj_cap_allprojects_dct])
+        assert type(loaded[0].project_scope) is AllProjectsScope
+
+    @pytest.mark.parametrize(
+        "capability",
+        [
+            EventsAcl([EventsAcl.Action.Read], scope=EventsAcl.Scope.All()),
+            EventsAcl([EventsAcl.Action.Write], scope=EventsAcl.Scope.DataSet([1, "2"])),
+            RawAcl([RawAcl.Action.Read], scope=RawAcl.Scope.Table({"my_db": ["my_table"]})),
         ],
     )
     def test_has_capability(
-        self, capabilities: ProjectCapabilitiesList, capability: Capability, expected: bool
+        self, proj_capabs_list: ProjectCapabilitiesList, project_name: str, capability: Capability
     ) -> None:
-        assert capabilities.includes_capability(capability) is expected
+        assert not proj_capabs_list.compare([capability], project=project_name)
+
+    @pytest.mark.parametrize(
+        "capability",
+        [
+            EventsAcl([EventsAcl.Action.Write], scope=EventsAcl.Scope.All()),
+            EventsAcl([EventsAcl.Action.Write], scope=EventsAcl.Scope.DataSet(["3"])),
+            EventsAcl([EventsAcl.Action.Write], scope=EventsAcl.Scope.DataSet([3])),
+            RawAcl([RawAcl.Action.Read], scope=RawAcl.Scope.Table({"my_db": ["unknown_table"]})),
+            RawAcl([RawAcl.Action.Read], scope=RawAcl.Scope.Table({"unknown_db": ["my_table"]})),
+        ],
+    )
+    def test_is_missing_capability(
+        self, proj_capabs_list: ProjectCapabilitiesList, project_name: str, capability: Capability
+    ) -> None:
+        missing_acls = proj_capabs_list.compare([capability], project=project_name)
+        assert missing_acls == [capability]
+
+        missing_acls = proj_capabs_list.compare([capability], project="do-es-nt-ex-is-ts")
+        assert missing_acls == [capability]
```

Comment:
Should we also have a test when you some of the capabilities? Input a set of capabilities, and some of them are missing?

Reply:
That's what this test does? Or am I misunderstanding your suggestion? 

Theres `test_has_capability` and `test_is_missing_capability`.

Reply:
Oh, a group input, yes, I can add that

Reply:
Will implement in another PR that makes comparisons simpler for e.g. ordinary capabilities.